enderio-machines/src/main/java/com/enderio/machines/common/menu/GhostMachineSlot.java

```diff
@@ -0,0 +1,49 @@
+package com.enderio.machines.common.menu;
+
+import com.enderio.machines.common.io.item.MachineInventory;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A ghost slot is a slot that can be populated with an item, but that item isn't really real.
+ * This item will also only have a stack size of one. It can never be more than 1.
+ * This item can be set or cleared in GUI but can never actually be "stolen" or deduct player resources.
+ */
+public class GhostMachineSlot extends MachineSlot {
```

Comment:
Did you also test it with "dragging"? I also made an implementation on my impulse hopper branch, and I needed a hack to fix it.

Reply:
Click and drag is covered, however shift clicking is not. I'll have a look at that when I get the chance :)